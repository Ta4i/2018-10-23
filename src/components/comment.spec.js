import React from 'react'
import { shallow } from 'enzyme'
import Comment from './comment'

describe('Comment', () => {
  const mockData = {
    user: 'test-user',
    text: 'some test text'
  }
  it('should render expected snapshot', function() {
    const wrapper = shallow(<Comment comment={mockData} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should have expected values', function() {
    const wrapper = shallow(<Comment comment={mockData} />)
    expect(wrapper.find('h4').text()).toEqual('test-user')
    expect(wrapper.find('p').text()).toEqual('some test text')
  })
})
