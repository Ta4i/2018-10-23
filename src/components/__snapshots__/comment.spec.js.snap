// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment should render expected snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comment
    comment={
      Object {
        "text": "some test text",
        "user": "test-user",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          test-user
        </h4>,
        <p>
          some test text
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test-user",
        },
        "ref": null,
        "rendered": "test-user",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "some test text",
        },
        "ref": null,
        "rendered": "some test text",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            test-user
          </h4>,
          <p>
            some test text
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test-user",
          },
          "ref": null,
          "rendered": "test-user",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some test text",
          },
          "ref": null,
          "rendered": "some test text",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
